### PR TITLE
Add sentry tracing

### PR DESCRIPTION
**Pull request by FelixMalfait**
Enable profiling, database tracing and graphql tracing

Tested on localhost:

Screenshot 2024-03-03 at 23 40 41

Screenshot 2024-03-03 at 23 41 24

Screenshot 2024-03-03 at 23 41 12

### CODE REVIEW

File: packages/twenty-server/src/integrations/tracing/useTracing.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        (o) => o.kind === Kind.OPERATION_DEFINITION,
+      ) as OperationDefinitionNode;
+      const operationType = rootOperation.operation;
+
+      if (Sentry.isInitialized()) {
+        Sentry.setTags({
+          operationName: transactionName,
+          operation: operationType,
+        });
+        Sentry.getCurrentScope().setTransactionName(transactionName);
```
Comment (FelixMalfait) [inline packages/twenty-server/src/integrations/tracing/useTracing.ts:21]: marked as deprecated but still works and used by most packages and might be un-deprecated soon? https://github.com/getsentry/sentry-javascript/issues/10846

Comment (FelixMalfait) [reply packages/twenty-server/src/integrations/tracing/useTracing.ts:21]: I've spent enough time on this so I'd say we go with this, not a big debt if we need to update it later! And it looks bad when we let sentry figure out the transaction name (everything under POST /graphql)

File: packages/twenty-server/src/integrations/tracing/useTracing.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import * as Sentry from '@sentry/node';
+import { handleStreamOrSingleExecutionResult, Plugin } from '@envelop/core';
+import { OperationDefinitionNode, Kind } from 'graphql';
+
+export const useTracing = (): Plugin => {
+  return {
+    onExecute({ args }) {
+      console.log('onExecute');
+      console.log(args);
```
Comment (magrinj) [inline packages/twenty-server/src/integrations/tracing/useTracing.ts:9]: Debug :)

File: packages/twenty-server/src/integrations/tracing/useTracing.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      const transactionName = args.operationName || 'Anonymous Operation';
+      const rootOperation = args.document.definitions.find(
+        (o) => o.kind === Kind.OPERATION_DEFINITION,
+      ) as OperationDefinitionNode;
+      const operationType = rootOperation.operation;
+
+      if (Sentry.isInitialized()) {
```
Comment (magrinj) [inline packages/twenty-server/src/integrations/tracing/useTracing.ts:16]: @FelixMalfait Maybe we should do something like `useExceptionHandler` and pass a service with custom drivers for tracing. So we can support multiple driver for tracing

File: packages/twenty-server/src/main.ts [hunk 4]
```diff
@@ -27,6 +28,11 @@ const bootstrap = async () => {
   // Use our logger
   app.useLogger(logger);
 
+  if (Sentry.isInitialized()) {
+    app.use(Sentry.Handlers.requestHandler());
+    app.use(Sentry.Handlers.tracingHandler());
+  }
```
Comment (magrinj) [inline packages/twenty-server/src/main.ts:34]: Is it really needed to make it works ?

Comment (FelixMalfait) [reply packages/twenty-server/src/main.ts:34]: Request Handler is apparently required for request scope isolation but I didn't dig into it too much, it's recommended. TracingHandler I'm guessing would guess starts the transaction, we could do that manually in the yoga plugin if we wanted eventually but the higher we start the better imo - so it feels right.

I've thought of doing something more abstract here where the service declares the providers but again I believe it's overkill and will make the code more complex without ever actually being useful

File: packages/twenty-server/src/integrations/tracing/useTracing.ts (hunk 3 above)
Comment (FelixMalfait) [reply packages/twenty-server/src/integrations/tracing/useTracing.ts:16]: I've considered it and started with this initially. There is a common driver for tracing which is called OpenTelemetry, and Sentry is implementing it. But there are several reasons why I deleted my implementation and went back to this. 
The 2 majors were that OpenTelemetry doesn't support profiling yet. And there is no case for enabling people to setup profiling on their own instance - this is really something specific to us. Also the instrumentation lib for Sentry pg worked best for me than the otel implementation.

**Review by magrinj — APPROVED** [review]
LGTM !

**Pull request merged**